core/primitives/src/block.rs

```diff
@@ -522,38 +521,59 @@ impl Block {
     }
 }
 
-#[derive(Clone)]
-pub enum MaybeNew<'a, T> {
-    New(&'a T),
-    Old(&'a T),
+#[derive(Clone, Debug)]
+pub enum ChunkType<'a> {
+    New(&'a ShardChunkHeader),
+    Old(&'a ShardChunkHeader),
 }
 
-fn annotate_chunk(
-    chunk: &ShardChunkHeader,
-    block_height: BlockHeight,
-) -> MaybeNew<ShardChunkHeader> {
-    if chunk.is_new_chunk(block_height) { MaybeNew::New(chunk) } else { MaybeNew::Old(chunk) }
+/// Implements Deref for ChunkType to allow access to ShardChunkHeader methods directly.
+impl Deref for ChunkType<'_> {
+    type Target = ShardChunkHeader;
+
+    fn deref(&self) -> &Self::Target {
+        match self {
+            ChunkType::New(chunk) => chunk,
+            ChunkType::Old(chunk) => chunk,
+        }
+    }
+}
+
+impl ChunkType<'_> {
+    pub fn is_new_chunk(&self) -> bool {
+        matches!(self, ChunkType::New(_))
+    }
 }
 
-pub enum ChunksCollection<'a> {
+// For BlockV1, we store the chunks in a Vec, else we use a slice reference.
+enum ChunksCollection<'a> {
     V1(Vec<ShardChunkHeader>),
     V2(&'a [ShardChunkHeader]),
 }
 
+/// Implements Deref for ChunksCollection to allow access to [ShardChunkHeader] methods directly.
+impl Deref for ChunksCollection<'_> {
```

Comment:
ditto

Reply:
This one is an internal struct, not exposed to public. This is just for convenience :)